internal/upstreamoidc/upstreamoidc.go

```diff
@@ -421,7 +421,7 @@ func maybeLogClaims(msg, name string, claims map[string]interface{}) {
 	}
 
 	if plog.Enabled(plog.LevelDebug) { // log keys at debug level
-		keys := sets.StringKeySet(claims).List() // note: this is only safe because the compiler asserts that claims is a map[string]<anything>
+		keys := psets.StringKeySet(claims).List() // note: this is only safe because the compiler asserts that claims is a map[string]<anything>
```

Comment:
Is this line not tested? Are we sure it is resilient at runtime to any valid input type?

Reply:
The original author added the comment which mentions that the `claims` parameter of the `maybeLogClaims` function is of type `map[string]interface{}`, which guarantees that it is a map with string keys, which is what is required by `StringKeySet`.

Also note that this line is not functionally changed. It still calls effectively the same code that it used to before we upgraded to Kube 0.26.

Unfortunately, it is not tested by the unit tests for that file. The integration tests execute that line many times, but they don't make assertions about the resulting log statements. Here is an example of a couple log statements resulting from running an integration test today:

```json
{"level":"debug","timestamp":"2023-01-19T16:56:45.616736Z","caller":"go.pinniped.dev/internal/upstreamoidc/upstreamoidc.go:425$upstreamoidc.maybeLogClaims","message":"claims from ID token","providerName":"test-upstream-oidc-idp-wrzmp","keys":["at_hash","aud","email","email_verified","exp","iat","iss","nonce","sub"]}
{"level":"debug","timestamp":"2023-01-19T16:56:45.617539Z","caller":"go.pinniped.dev/internal/upstreamoidc/upstreamoidc.go:425$upstreamoidc.maybeLogClaims","message":"claims from ID token and userinfo","providerName":"test-upstream-oidc-idp-wrzmp","keys":["at_hash","aud","email","email_verified","exp","iat","iss","nonce","sub"]}
```

Reply:
Resolving conversation since we know that the code path is executed during integration tests.